### PR TITLE
fix: don't sent device name if DefaultPII is disabled

### LINKED ISSUE #2063: `DefaultAndroidEventProcessor#getDevice` should not set `getDeviceName` if `sendDefaultPii` isn't enabled
### Description

`getDeviceName` could return a name defined by the user and this could lead to PII.

* Bump the Android SDK for the Flutter SDK

### PR DESCRIPTION
**Pull request by vaind**
## :bulb: Motivation and Context

closes #2063

## :pencil: Checklist

- [x] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java [hunk 1]
```diff
@@ -282,7 +282,9 @@ private void setArchitectures(final @NotNull Device device) {
     // TODO: missing usable memory
 
     Device device = new Device();
-    device.setName(getDeviceName());
+    if (!options.isSendDefaultPii()) {
```
Comment (adinauer) [inline sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java:285]: Shouldn't this be the other way around? i.e. only set the device name if `sendDefaultPii` is enabled?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ We suggest opening an issue to discuss bigger changes before investing on a big
 
 # Requirements
 
-The project currently requires you run JDK 11.
+The project currently requires you run JDK 17.
```
Comment (adinauer) [inline CONTRIBUTING.md:8]: 🙏

**Review by adinauer — CHANGES_REQUESTED** [review]
Logic seems flipped. Could you please also add an entry to the changelog?

File: sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java (hunk 1 above)
Comment (vaind) [reply sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java]: 🤦 was a long week...

**Review by adinauer — APPROVED** [review]

File: sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java [hunk 3]
```diff
@@ -94,7 +94,7 @@ public DefaultAndroidEventProcessor(
 
     ExecutorService executorService = Executors.newSingleThreadExecutor();
     // dont ref. to method reference, theres a bug on it
-    contextData = executorService.submit(() -> loadContextData());
+    contextData = executorService.submit(this::loadContextData);
```
Comment (romtsn) [inline sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java:97]:
```suggestion
    contextData = executorService.submit(() -> loadContextData());
```
there's a comment above ;)

**Review by romtsn — APPROVED** [review]

File: sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java (hunk 3 above)
Comment (vaind) [reply sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java]: yeah, I just noticed that too after going over it one last time. disabled the linter for the line

**Pull request merged**